src/Workspaces/Core/Desktop/Workspace/SQLite/SQLitePersistentStorage.cs

```diff
@@ -243,6 +243,10 @@ public override void Initialize(Solution solution)
             {
                 var connection = pooledConnection.Connection;
 
+                // Configure journal and locking mode
```

Comment:
does pragma always affect whole SQLite or one specific connection it is executed?

Reply:
if user close VS or kill VS through task manager without closing solution (a lot of people do that), with this
"connection.ExecuteCommand("pragma journal_mode=memory", throwOnError: false);" will SQLite be in corrupted state (which is fine, it happened when we used Esent as well), if it does, will it recover the corrupted state when next time VS opens the same solution? (this is important since 8 out of 10 db is in corrupted state, according to data I saw from esent, and if it can't recover it needs to rebuild from scratch and that is basically same as not having persisted storage)

Reply:
As far as I understand how SQLite works, locking mode is set for a connection, but exclusive mode never unlocks the database until this connection is closed (as SQLite docs say: "When the locking-mode is set to EXCLUSIVE, the database connection never releases file-locks. The first time the database is read in EXCLUSIVE mode, a shared lock is obtained and held. The first time the database is written, an exclusive lock is obtained and held.").
This would normally pose a problem (one connection locking out all other connections), but since shared cache is used, the normal locking rules [do not apply](https://www.sqlite.org/sharedcache.html#shared_cache_locking_model). This is to be tested though.

Reply:
@heejaechang This is why I originally proposed `wal` instead of `memory` -- crash-resistance. I think that we can do some test with heavy write scenarios (delete both dbs, load Roslyn solution w/PerfView enabled, wait till it rebuilds all dbs, analyze trace). I [predict](https://github.com/dotnet/roslyn/pull/22340#issuecomment-334260782) that the difference between `memory` and `wal` would be minimal, especially in exclusive mode (WAL doesn't need SHM in exclusive mode).